langchain/src/vectorstores/cloudflare_vectorize.ts

```diff
@@ -0,0 +1,227 @@
+import * as uuid from "uuid";
+
+import {
+  VectorizeIndex,
+  VectorizeVectorMetadata,
+} from "@cloudflare/workers-types";
+import { VectorStore } from "./base.js";
+import { Embeddings } from "../embeddings/base.js";
+import { Document } from "../document.js";
+import { chunkArray } from "../util/chunk.js";
+import { AsyncCaller, type AsyncCallerParams } from "../util/async_caller.js";
+
+export interface VectorizeLibArgs extends AsyncCallerParams {
+  index: VectorizeIndex;
+  textKey?: string;
+}
+
+/**
+ * Type that defines the parameters for the delete operation in the
+ * CloudflareVectorizeStore class. It includes ids, deleteAll flag, and namespace.
+ */
+export type VectorizeDeleteParams = {
+  ids: string[];
+};
+
+/**
+ * Class that extends the VectorStore class and provides methods to
+ * interact with the Cloudflare Vectorize vector database.
+ */
+export class CloudflareVectorizeStore extends VectorStore {
+  textKey: string;
+
+  namespace?: string;
+
+  index: VectorizeIndex;
+
+  caller: AsyncCaller;
+
+  _vectorstoreType(): string {
+    return "cloudflare_vectorize";
+  }
+
+  constructor(embeddings: Embeddings, args: VectorizeLibArgs) {
+    super(embeddings, args);
+
+    this.embeddings = embeddings;
+    const { index, textKey, ...asyncCallerArgs } = args;
+    if (!index) {
+      throw new Error(
+        "Must supply a Vectorize index binding, eg { index: env.VECTORIZE }"
+      );
+    }
+    this.index = index;
+    this.textKey = textKey ?? "text";
+    this.caller = new AsyncCaller({
+      maxConcurrency: 6,
+      maxRetries: 0,
```

Comment:
No retries?

Reply:
Not on the internal bindings – or at least not for now. It was retrying on user errors, eg putting the wrong thing in metadata, etc. These would be 400 errors if they were HTTP. Would need to rework the retry mechanism to deal with these (and other) custom errors.